processing/src/main/java/org/apache/druid/segment/incremental/IncrementalIndexStorageAdapter.java

```diff
@@ -236,9 +236,9 @@ public Metadata getMetadata()
     {
       currEntry = new IncrementalIndexRowHolder();
       columnSelectorFactory = new IncrementalIndexColumnSelectorFactory(index, virtualColumns, descending, currEntry);
+      maxRowIndex = index.getLastRowIndex();
```

Comment:
Would you please add a comment about this must be done before creating the below filter?

Reply:
Added a comment